src/main/java/net/dv8tion/jda/internal/audio/AudioConnection.java

```diff
@@ -663,8 +666,8 @@ public ByteBuffer getNextPacketRaw(boolean changeTalking)
                 cond: if (sentSilenceOnConnect && sendHandler != null && sendHandler.canProvide())
                 {
                     silenceCounter = -1;
-                    byte[] rawAudio = sendHandler.provide20MsAudio();
-                    if (rawAudio == null || rawAudio.length == 0)
+                    ByteBuffer rawAudio = sendHandler.provide20MsAudio();
+                    if (rawAudio == null || !rawAudio.hasRemaining() || !rawAudio.hasArray()) // no array = cannot encrypt
```

Comment:
`hasRemaining` can be true when there is no array?

Reply:
Yes, no array just means its not a heap buffer but a direct buffer. `hasRemaining()` is a short handle for `position() < limit()`.